backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py

```diff
@@ -536,6 +627,143 @@ def search_content(
+        search = search.query("simple_query_string", query=search_string, fields=fields)
+        known_user_ids = UserApi(
+            current_user=None, session=self._session, config=self._config
+        ).get_known_user_ids(self._user.user_id)
+        search = search.filter("terms", user_id=known_user_ids)
+        if not show_active:
+            search = search.exclude("term", is_active=True)
+        if not show_deleted:
+            search = search.exclude("term", is_deleted=True)
+        if workspace_ids:
+            search = search.filter("terms", workspace_ids=workspace_ids)
+        newest_authored_content_date_range = self.create_es_datetime_range(
+            newest_authored_content_date_from, newest_authored_content_date_to
+        )
+        if newest_authored_content_date_range:
+            search = search.filter(
+                "range", newest_authored_content_date=newest_authored_content_date_range
+            )
+
+        if size:
+            search = search.extra(size=size)
+        if page_nb:
+            search = search.extra(from_=self.offset_from_pagination(size, page_nb))
+
+        search.aggs.bucket("workspace_ids", "terms", field="workspace_ids")
+        search.aggs.metric(
+            "newest_authored_content_date_from", "min", field="newest_authored_content_date",
+        )
+        search.aggs.metric(
+            "newest_authored_content_date_to", "max", field="newest_authored_content_date",
+        )
+        response = search.execute()
+        known_workspaces = self._get_workspaces_known_to_user()
+        facets = {
+            "workspaces": self._create_filtered_facets(
+                "workspace_id", response.aggregations.workspace_ids.buckets, known_workspaces
+            )
+        }
+        return UserSearchResponse(
+            hits=response["hits"],
+            facets=facets,
+            search_fields=search_fields,
+            newest_authored_content_date_from=response.aggregations.newest_authored_content_date_from,
+            newest_authored_content_date_to=response.aggregations.newest_authored_content_date_to,
+        )
+
+    def search_workspace(
+        self,
+        search_string: str,
+        search_fields: typing.Optional[typing.List[str]] = DEFAULT_WORKSPACE_SEARCH_FIELDS,
+        member_ids: typing.Optional[typing.List[int]] = None,
+        show_deleted: bool = False,
+        page_nb: int = 0,
+        size: int = 10,
+    ) -> WorkspaceSearchResponse:
+        search = Search(
+            using=self.es,
+            doc_type=IndexedWorkspace,
+            index=self._get_index_parameters(IndexedWorkspace).alias,
+        )
+
+        fields = [
+            "label.exact^5",
+            "label^3",
+            "description^3",
+        ]
+        search_fields = search_fields or DEFAULT_WORKSPACE_SEARCH_FIELDS
+        fields = [field for field in fields if name_starts_with_any_prefix(field, search_fields)]
```

Comment:
similar advice: should be consistent with content sooner or later.

Reply:
Done.